### PR TITLE
feat(ethereum contracts): Introduce `BridgingPayment` for ethereum

### LINKED ISSUE #80: Bridging payment contract on ethereum

### PR DESCRIPTION
**Pull request by mertwole**
Resolves #80

### CODE REVIEW

File: ethereum/src/BridgingPayment.sol [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        underlying = _underlying;
+        admin = _admin;
+        fee = _fee;
+    }
+
+    /** @dev Deduct a `fee` from the user and transfer it to the `admin` address.
+     *  This function reverts if user don't have enough funds to pay the fee.
+     */
+    function deductFee() internal {
+        (bool feeTransferSuccess, ) = admin.call{value: fee}("");
```
Comment (LouiseMedova) [inline ethereum/src/BridgingPayment.sol:24]: For sending Ether here, I would use transfer:
https://dev.to/tchisom17/understanding-ether-transfers-in-solidity-send-transfer-and-call-3k67#:~:text=Your%20smart%20contract%20in%20Solidity,to%20handle%20transfer%20failures%20manually.

File: ethereum/src/ERC20Gateway.sol [hunk 2]
```diff
@@ -72,3 +73,24 @@ contract ERC20Gateway is IERC20Gateway, Context, IMessageQueueReceiver {
...
+
+    /** @dev Call `requestBridging` function from `ERC20Gateway` contract. This function also
+     * deducting some fee in native tokens from such a call. For further info see `ERC20Gateway::requestBridging`.
+     */
+    function requestBridging(
+        address token,
+        uint256 amount,
+        bytes32 to
+    ) public payable {
+        deductFee();
```
Comment (LouiseMedova) [inline ethereum/src/ERC20Gateway.sol:92]: While it is very unlikely due to the logic (a call is made to the admin's address), a reentrancy attack is still possible

Comment (mertwole) [reply ethereum/src/ERC20Gateway.sol:92]: Is it resolved by using `transfer` instead of `call`?

Comment (LouiseMedova) [reply ethereum/src/ERC20Gateway.sol:92]: Yes

**Review by LouiseMedova — APPROVED** [review]

**Pull request merged**